### DISCUSSION TITLE
Is there a way to store systems in Vec?

### DISCUSSION THREAD

**Discussion by Nigless** [bevyengine/bevy#13926] [General]
![image](https://github.com/bevyengine/bevy/assets/40428432/0c6f65c5-3b2d-4c33-adb6-e05458a150ea)

**Comment by bugsweeper** [comment]
I think there is no way, because Box is owning, but not shared smartpointer, IntoSystemConfigs takes it by value, so you can't keep it after add_systems, you can just move it.

**Reply by bugsweeper** [reply]
You can achieve partial goal by keeping specific functions
```
#[derive(Component, Reflect, Default)]
#[reflect(Component)]
pub struct StandingState;

#[derive(Default)]
pub struct StandingStatePlugin {
    enter: Vec<fn(Commands, Res<AssetServer>) -> ()>,
}

impl Plugin for StandingStatePlugin {
    fn build(&self, app: &mut App) {
        app.register_type::<StandingState>().add_systems(
            PreUpdate,
            (IntoSystem::into_system(self.enter[0]),).in_set(GhostSystems::Enter),
        );
    }
}

```

**Reply by bugsweeper** [reply]
That because System can be made from functions and from closures, functions don't keep any values, that is why is clonable, closure can keep not clonable values, that is why it's not clonable.

**Comment by NiklasEi** [comment]
In bevy_asset_loader I do this with `Vec<SystemConfig>`.

https://github.com/NiklasEi/bevy_asset_loader/blob/e7c389dec6f51f3a38a66d25367db7010bc743f5/bevy_asset_loader/src/loading_state/config.rs#L88-L97

**Reply by bugsweeper** [reply]
But you move `SystemConfig`'s into `add_systems` instead of just keeping it in `Vec` after `add_systems`

**Reply by Nigless** [reply]
I dont need to keep it

**Reply by Nigless** [reply]
I cant move it out of vector 
![image](https://github.com/bevyengine/bevy/assets/40428432/faf249db-104e-47da-95d1-9c5961ca4f8a)

**Reply by NiklasEi** [reply]
If you are ok with the vector being empty after the loop, you can use `.drain(..)` on the vecs like:
```rust
for system in self.enter,drain(..) {
    app.add_systems(PreUpdate, system.in_set(StateMachineSystems::Enter));
}
```
